regexes/oss.yml

```diff
@@ -429,6 +429,46 @@
 - regex: '(?:(?:Orca-)?(?<!like |/|RadioPublic |Anghami )Android|Adr|AOSP)[ /]?(?:[a-z]+ )?(\d+[\.\d]*)'
   name: 'Android'
   version: '$1'
+  versions:
+    - regex: 'Android[ /]34(?!.)'
```

Comment:
create a separate rule above
```yml
- regex: 'Android[ /]\d{2}$'
  name: 'Android'
  versions:
    - regex: 'Android[ /]34'
      version: '14'
````
etc...
this will reduce the number of iterations in the loop.
if you look at the code, you will understand everything:

https://github.com/matomo-org/device-detector/blob/59fb555c303eac7cd8fe6e9d0a56347faa2a3ad5/Parser/OperatingSystem.php#L508-L534

Reply:
After reading your comment, I realized I could combine the sdk version blocks, and lead to better code overall. That was not easy, some of the versions it is almost down to looking to see what range the version is in, but I think I succeeded without writing any php.